lib/ConstExtract/ConstExtract.cpp

```diff
@@ -119,17 +120,58 @@ parseProtocolListFromFile(StringRef protocolListFilePath,
   return true;
 }
 
+static std::string extractLiteralOutput(Expr *expr) {
+  std::string LiteralOutput;
+  llvm::raw_string_ostream OutputStream(LiteralOutput);
+  expr->printConstExprValue(&OutputStream, nullptr);
+
+  return LiteralOutput;
+}
+
 static std::shared_ptr<CompileTimeValue>
 extractPropertyInitializationValue(VarDecl *propertyDecl) {
   auto binding = propertyDecl->getParentPatternBinding();
   if (binding) {
     auto originalInit = binding->getOriginalInit(0);
     if (originalInit) {
-      std::string LiteralOutput;
-      llvm::raw_string_ostream OutputStream(LiteralOutput);
-      originalInit->printConstExprValue(&OutputStream, nullptr);
-      if (!LiteralOutput.empty())
-        return std::make_shared<RawLiteralValue>(LiteralOutput);
+      auto literalOutput = extractLiteralOutput(originalInit);
+      if (!literalOutput.empty()) {
+        return std::make_shared<RawLiteralValue>(literalOutput);
+      }
+
+      if (auto callExpr = dyn_cast<CallExpr>(originalInit)) {
+        std::vector<FunctionParameter> parameters;
+
+        const auto args = callExpr->getArgs();
+        for (auto arg : *args) {
+          auto label = arg.getLabel().str().str();
+          auto expr = arg.getExpr();
+
+          switch (expr->getKind()) {
+          case ExprKind::DefaultArgument: {
+            auto defaultArgument = cast<DefaultArgumentExpr>(expr);
+            auto *decl = defaultArgument->getParamDecl();
+
+            if (decl->hasDefaultExpr()) {
+              literalOutput =
+                  extractLiteralOutput(decl->getTypeCheckedDefaultExpr());
+            }
+
+            break;
+          }
+          default:
+            literalOutput = extractLiteralOutput(expr);
+            break;
+          }
+
+          if (!literalOutput.empty()) {
+            parameters.push_back({label, expr->getType(), literalOutput});
+          }
+        }
+
+        auto name = toFullyQualifiedTypeNameString(callExpr->getType());
+        return std::make_shared<InitCallValue>(name, parameters);
```

Comment:
Hmm, we are not doing any kind of verification to ensure that what we have is actually an `init` call. 

We have a guarantee that `toFullyQualifiedTypeNameString(callExpr->getType())` returns the same thing that is the type of this property, but this can be any old function that returns this type. And if this is some `foo(_ arg: String) -> Int`, then this isn't the same as an `init` call on this property's type, and is not useful to clients, I imagine. 

I don't remember off the top of my head how to check that the function being called is an `init`, but that's something worth looking into. We make sure to specify that we are collecting `InitCallValue`, so if it's some other function we are calling here, I think it would make sense to leave the entire property's value to be `RuntimeValue`.

Reply:
Great catch!

I added a check for `ExprKind::ConstructorRefCall` to ensure we are only handling init calls. Added a test to capture this negative case.